pyccel/ast/builtin_methods/set_methods.py

```diff
@@ -132,3 +132,30 @@ def __init__(self, set_variable):
         self._precision = set_variable._precision
         self._class_type = set_variable._class_type
         super().__init__(set_variable)
+
+
+class SetPop(SetMethod):
+    """
+    Represents a call to the .pop() method.
+
+    The pop() method pops an element from the set. 
+    It does not take any argument but returns the popped 
+    element. It raises an error if the element is not 
+    present in the set.
+
+    Parameters
+    ----------
+    set_variable : TypedAstNode
+        The name of the set.
+    """
+    __slots__ = ('_dtype', '_precision', '_class_type')
+    _rank = 0
+    _order = None
+    _shape = None
+    name = 'pop'
+
+    def __init__(self, set_variable):
+        self._dtype = set_variable.dtype
+        self._precision = set_variable.precision
+        self._class_type = set_variable.class_type
```

Comment:
Are you sure about the `class_type` here?

Reply:
I think it should be  `self._class_type = set_variable.dtype` because the set in Pyccel contains just hash scalar types (without tuples).

Reply:
Fixed : 21c64ad